apps/backoffice-v2/src/lib/blocks/hooks/useEntityInfoBlock/useEntityInfoBlock.tsx

```diff
@@ -0,0 +1,136 @@
+import { generateEditableDetailsV2Fields } from '@/common/components/organisms/EditableDetailsV2/utils/generate-editable-details-v2-fields';
+import { useAuthenticatedUserQuery } from '@/domains/auth/hooks/queries/useAuthenticatedUserQuery/useAuthenticatedUserQuery';
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { useEventMutation } from '@/domains/workflows/hooks/mutations/useEventMutation/useEventMutation';
+import { useUpdateContextAndSyncEntityMutation } from '@/domains/workflows/hooks/mutations/useUpdateContextAndSyncEntity/useUpdateContextAndSyncEntity';
+import { createBlocksTyped } from '@/lib/blocks/create-blocks-typed/create-blocks-typed';
+import { useCaseState } from '@/pages/Entity/components/Case/hooks/useCaseState/useCaseState';
+import { isObject, valueOrNA } from '@ballerine/common';
+import { useCallback, useMemo } from 'react';
+import { titleCase, toTitleCase } from 'string-ts';
+
+export const useEntityInfoBlock = ({
+  entity,
+  workflow,
+  entityDataAdditionalInfo,
+}: {
+  entity: TWorkflowById['context']['entity'];
+  workflow: TWorkflowById;
+  entityDataAdditionalInfo: TWorkflowById['context']['entity']['data']['additionalInfo'];
+}) => {
+  const { mutate: mutateEvent } = useEventMutation();
+  const onMutateEvent = useCallback(() => {
+    mutateEvent({
+      workflowId: workflow?.id,
+      event: 're_run_checks',
```

Comment:
a bit more explicit, tomorrow there will be another block that will run different checks